### ISSUE TITLE
Could we build a Litentry OAuth server based on this

### ISSUE THREAD

**Issue by markdavison** [litentry/PolkaSignIn#5]
Would be cool if we could use this as the basis for an OAuth server to allow easy integration for web 2.0 apps. 

We could then submit PR's to packages like https://github.com/simov/grant/blob/master/config/oauth.json to make adding authentication using your wallet easy for website owners to add.

**Comment by hanwencheng** [comment]
Both sounds good to me.

**Comment by hanwencheng** [comment]
The first one is meaningful in order to:
1. release the developers from building the backend services.
2. provide a trusted backend to provide challenges (still need to prevent the man in the middle attack by signing the challenge with a server-side private key.)
3. Work as an example for external parties to build a customized OAuth authorization server for their own federated services.

**Comment by hanwencheng** [comment]
@markdavison can you work on this issue?

**Comment by markdavison** [comment]
@hanwencheng I am currently working on the evm indexing. Is this higher priority?

**Comment by hanwencheng** [comment]
No, it is not of high priority, how many hours do you think this is needed in both F/E and B/E?

**Comment by markdavison** [comment]
I think there will be a very small amount of FE. Should just be the authorisation page. It should be mostly BE. 

I think its best if I do a spike into how it might work.. maybe half a day? Then I should be able to give more accurate estimates around exactly what is involved.

**Comment by markdavison** [comment]
I did some digging on this last week. Looks like about a days worth of B/E and probably half a day of F/E. Then we would need to host it.. so half a day of dockerising it and deploying it to EC2.

I guess a separate piece would be to raise some PR's to some popular oAuth libs to make it easy for developers to add 'sign-in with PolkaSignIn'.

**Comment by hanwencheng** [comment]
`so half a day of dockerising it and deploying it to EC2.` this is included in one day of B/E, correct?

**Comment by markdavison** [comment]
Sorry, no that was a separate devops task. 

So ~1 day B/E, ~0.5 day F/E and ~0.5 day of devops. 

Then probably some extra time to raise some PR's to other OAuth packages to make integration really easy for developers.

**Comment by hanwencheng** [comment]
I will follow up on the PolkaSignIn next week.
